### PR TITLE
Implement addExtensionsTags in OsmGpxWriteContext

### PR DESCRIPTION
**Pull request by RZR-UA**

### CODE REVIEW

**Comment by RZR-UA** [comment]
https://github.com/osmandapp/OsmAnd/issues/18866

**Review by ivanPyrohivskyi — APPROVED** [review]

File: java-tools/OsmAndMapCreatorUtilities/src/main/java/net/osmand/obf/OsmGpxWriteContext.java [hunk 1]
```diff
@@ -189,6 +191,24 @@ public void writeTrack(OsmGpxFile gpxInfo, Map<String, String> extraTrackTags, G
...
+		List<String> pgIcons = new ArrayList<>();
+		List<String> pgColors = new ArrayList<>();
+		List<String> pgBackgrounds = new ArrayList<>();
+		for (String name : pointsGroups.keySet()) {
+			pgNames.add(name);
+			pgIcons.add(pointsGroups.get(name).iconName);
+			pgBackgrounds.add(pointsGroups.get(name).backgroundType);
+			pgColors.add(Algorithms.colorToString(pointsGroups.get(name).color));
+		}
+		final String DELIMITER = "~~~";
```
Comment (ivanPyrohivskyi) [inline java-tools/OsmAndMapCreatorUtilities/src/main/java/net/osmand/obf/OsmGpxWriteContext.java:205]: DELIMITER in generation and Android must be same constant (Kotlin lib)

File: java-tools/OsmAndMapCreatorUtilities/src/main/java/net/osmand/obf/OsmGpxWriteContext.java [hunk 2]
```diff
@@ -189,6 +191,24 @@ public void writeTrack(OsmGpxFile gpxInfo, Map<String, String> extraTrackTags, G
...
+		gpxTrackTags.put("points_groups_names", String.join(DELIMITER, pgNames));
```
Comment (ivanPyrohivskyi) [inline java-tools/OsmAndMapCreatorUtilities/src/main/java/net/osmand/obf/OsmGpxWriteContext.java:206]: points_groups_names must be in constant in Kotlin lib

File: java-tools/OsmAndMapCreatorUtilities/src/main/java/net/osmand/obf/OsmGpxWriteContext.java [hunk 3]
```diff
@@ -187,6 +191,41 @@ public void writeTrack(OsmGpxFile gpxInfo, Map<String, String> extraTrackTags, G
...
+				gpxTrackTags.put("color", extensions.get("color")); // osmand:color
+				// prioritize osmand:color over GPX color
+				gpxTrackTags.remove("colour_int");
+				gpxTrackTags.remove("colour");
+			}
+			if (extensions.containsKey("width")) {
+				gpxTrackTags.put("width", extensions.get("width")); // osmand:width
+			}
+			for (String key : extensions.keySet()) {
+				gpxTrackTags.put("gpx_" + key, extensions.get(key)); // enlist required tags in poi_types.xml
```
Comment (ivanPyrohivskyi) [inline java-tools/OsmAndMapCreatorUtilities/src/main/java/net/osmand/obf/OsmGpxWriteContext.java:224]: GPX_EXTENSION_TAG_PREFIX - common for generation and Android/iOS

**Review by ivanPyrohivskyi — APPROVED** [review]

**Comment by RZR-UA** [comment]
Please do not forget to merge child request https://github.com/osmandapp/OsmAnd-tools/pull/895 before merge current one into master.

**Pull request merged**